acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -52,9 +49,87 @@ def kill_pcp_broker(host)
   end
 end
 
+# Query pcp-broker's inventory of associated clients
+# Reference: https://github.com/puppetlabs/pcp-specifications/blob/master/pcp/inventory.md
+# @param broker hostname or beaker host object of the pcp-broker host
+# @param query pattern of client identities that you want to check e.g. pcp://*/agent for all agents, or pcp://*/* for everything
+# @return Ruby array of strings that are the client identities that match the query pattern
+# @raise Exception if pcp-client cannot connect to make the inventory request, does not receive a response, or the response is missing
+#        the required [:data]["uris"] property
+def pcp_broker_inventory(broker, query)
+  # Event machine is required by the ruby-pcp-client gem
+  # https://github.com/puppetlabs/ruby-pcp-client
+  em = Thread.new { EM.run }
+
+  mutex = Mutex.new
+  have_response = ConditionVariable.new
+  response = nil
+
+  client = connect_pcp_client(broker)
+
+  client.on_message = proc do |message|
+    mutex.synchronize do
+      resp = {
+        :envelope => message.envelope,
+        :data     => JSON.load(message.data),
+        :debug    => JSON.load(message.debug)
+      }
+      response = resp
+      have_response.signal
+    end
+  end
+
+  message = PCP::Message.new({
+    :message_type => 'http://puppetlabs.com/inventory_request',
+    :targets => ["pcp:///server"]
+  })
+
+  message.data = {
+    :query => [query]
+  }.to_json
+
+  message_expiry = 10 # Seconds for the PCP message to be considered failed
+  inventory_expiry = 60 # Seconds for the inventory request to be considered failed
+  message.expires(message_expiry)
+
+  client.send(message)
+
+  begin
+    Timeout::timeout(inventory_expiry) do
+      done = false
+      loop do
+        mutex.synchronize do
+          have_response.wait(mutex)
+        end
+        break if response
+      end
+    end
+  rescue Timeout::Error
+    raise "Didn't receive a response for PCP inventory request"
+  end # wait for message
+
+  em.exit
```

Comment:
This isn't particularly safe.  Best to call EM.stop and let the thread exit normally.